flipperdevices/flipperzero-firmware
# [FL-1800] IRDA: enc/decoder refactoring, Add NEC42

# What's new

- Added NEC42
- NECext decode rules mitigation
- Refactoring: support multi-length protocols in common part

# Verification 

- Try read NEC42 remotes (Knopkus has it)
- Try read NECext remotes (Knopkus has it)
- Also check that common NEC works same (it's already checked by unit tests, but it wouldn't do harm to test it manually too)

# Checklist (do not modify)
